drivers/scheduler/rke/rke.go

```diff
@@ -1,28 +1,234 @@
+		log.Fatalf(fmt.Sprintf("%v", err))
+	}
+	pxLabel := make(map[string]string)
+	pxLabel[PxLabelNameKey] = PxLabelValue
+	pxPodRestartCountMap, err := k.GetPodsRestartCount(namespace, pxLabel)
+	if err != nil {
+		log.Fatalf(fmt.Sprintf("%v", err))
+	}
+	for pod, value := range pxPodRestartCountMap {
+		n, err := node.GetNodeByIP(pod.Status.HostIP)
+		if err != nil {
+			log.Fatalf(fmt.Sprintf("%v", err))
+		}
+		n.PxPodRestartCount = value
+	}
+	k.SpecFactory, err = spec.NewFactory(scheduleOpts.SpecDir, scheduleOpts.VolDriverName, k)
+	if err != nil {
+		return err
+	}
+	err, RancherClusterParametersValue = k.GetRancherClusterParametersValue()
+	if err != nil {
+		return err
+	}
+	rancherClientOpts := rancherClientBase.ClientOpts{
+		URL:       RancherClusterParametersValue.Endpoint,
+		TokenKey:  RancherClusterParametersValue.Token,
+		AccessKey: RancherClusterParametersValue.AccessKey,
+		SecretKey: RancherClusterParametersValue.SecretKey,
+		Insecure:  true,
+	}
+	k.client, err = rancherClient.NewClient(&rancherClientOpts)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetRancherClusterParametersValue returns the rancher token,endpoint,secret key, access key
+func (k *Rancher) GetRancherClusterParametersValue() (error, *RancherClusterParameters) {
+	var data map[string]interface{}
+	var var1 RancherClusterParameters
+	// To Do: Rancher URL for cloud cluster will not be fetched from master node IP
+	masterNodeName := node.GetMasterNodes()[0].Name
+	endpoint := "https://" + masterNodeName + "/v3"
+	rancherURL := "https://" + masterNodeName + "/v3-public/localProviders/local?action=login"
+	//To get from config map after https://github.com/portworx/torpedo/pull/1517 is merged
+	username := "admin"
+	password := "1JVWD8juKPB8fVgK"
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+	body := strings.NewReader(fmt.Sprintf(`{"username":"%s","password":"%s"}`, username, password))
+	req, err := http.NewRequest("POST", rancherURL, body)
+	if err != nil {
+		log.Errorf("Failed to create rancher POST request:", err)
+		return err, &var1
+	}
+	req.Header.Add("Content-Type", "application/json")
+	resp, err := client.Do(req)
+	if err != nil {
+		fmt.Println("Failed to send request:", err)
+		return err, &var1
+	}
+	defer resp.Body.Close()
+	respBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Println("Failed to read response:", err)
+		return err, &var1
+	}
+	response := string(respBody)
+	err = json.Unmarshal([]byte(response), &data)
+	if err != nil {
+		fmt.Println("Error:", err)
+		return err, &var1
+	}
+	var1.Endpoint = endpoint
+	var1.Token = data["token"].(string)
+	var1.AccessKey = strings.Split(data["token"].(string), ":")[0]
+	var1.SecretKey = strings.Split(data["token"].(string), ":")[1]
+	return nil, &var1
+}
+
+// GetActiveRancherClusterId returns the ID of active rancher cluster
+func (k *Rancher) GetActiveRancherClusterId() (string, error) {
```

Comment:
How about GetActiveRancherCluster"ID" to be consistent?

Reply:
done